src/pool/mod.rs

```diff
@@ -87,30 +87,34 @@ impl Pool {
     pub async fn simulate_swap_mut<P: 'static + JsonRpcClient>(
         &mut self,
         token_in: H160,
-        amount_in: u128,
+        amount_in: U256,
         provider: Arc<Provider<P>>,
-    ) -> Result<u128, CFFMError<P>> {
+    ) -> Result<U256, CFFMError<P>> {
         match self {
             Pool::UniswapV2(pool) => Ok(pool.simulate_swap_mut(token_in, amount_in)),
             Pool::UniswapV3(pool) => pool.simulate_swap_mut(token_in, amount_in, provider).await,
         }
     }
 }
 
-fn convert_to_decimals(amount: u128, decimals: u8, target_decimals: u8) -> u128 {
+fn convert_to_decimals(amount: U256, decimals: u8, target_decimals: u8) -> U256 {
     match target_decimals.cmp(&decimals) {
-        Ordering::Less => amount * 10u128.pow((decimals - target_decimals) as u32),
-        Ordering::Greater => amount * 10u128.pow((target_decimals - decimals) as u32),
-        Ordering::Equal => amount,
+        Ordering::Less => {
+            U256::from((amount).as_u128() / 10u128.pow((decimals - target_decimals) as u32))
+        }
+        Ordering::Greater => {
+            U256::from((amount).as_u128() * 10u128.pow((target_decimals - decimals) as u32))
+        }
+        _ => amount,
```

Comment:
If `amount` is a `U256`, calling `as_u128()` on it could cause a panic if the `amount` is greater than `u128::MAX`. 

Can you change this to be something like
```rust
fn convert_to_decimals(amount: U256, decimals: u8, target_decimals: u8) -> U256 {
    match target_decimals.cmp(&decimals) {
        Ordering::Less => amount / U256::from(10u128.pow((decimals - target_decimals) as u32)),
        Ordering::Greater => amount * U256::from(10u128.pow((target_decimals - decimals) as u32)),
        _ => amount,
    }
}
```